src/optimize/base_optimizer.js

```diff
@@ -369,7 +369,34 @@ export default class BaseOptimizer {
           parallel: true,
           sourceMap: false,
           uglifyOptions: {
-            compress: false,
+            compress: {
+              // the following is required for dead-code the removal
+              // check in React DevTools
+
+              unused: true,
+              dead_code: true,
+              conditionals: true,
+              evaluate: true,
+
+              comparisons: false,
+              sequences: false,
+              properties: false,
+              drop_debugger: false,
+              booleans: false,
+              loops: false,
+              toplevel: false,
+              top_retain: false,
+              hoist_funs: false,
+              if_return: false,
+              join_vars: false,
+              collapse_vars: false,
+              reduce_vars: false,
+              warnings: false,
+              negate_iife: false,
+              keep_fnames: true,
+              keep_infinity: true,
```

Comment:
I kinda wish all the `true`s were together at the top or something, but I don't care

Reply:
These are to disable functionality enabled by default in the compressor. The UglifyJS options aren't really consistent in that respect. https://github.com/mishoo/UglifyJS2/tree/v2.x